src/sugar3/graphics/popwindow.py

```diff
@@ -0,0 +1,146 @@
+from gi.repository import GdkX11
+
+from sugar3.graphics import style
+from sugar3.graphics.toolbutton import ToolButton
+from sugar3.graphics.toolbarbox import ToolbarBox
+
+
+class PopWindow(Gtk.Window):
+    """
+    UI interface for activity Pop-up Windows
+
+    """
+    def __init__(self, title=None, set_titlebox=True, window_xid=None):
+        Gtk.Window.__init__(self)
+        self._title = title
+        self._set_title_box = set_titlebox
+        self._parent_window_xid = window_xid
+
+        self._decorated = False
+        self._position = Gtk.WindowPosition.CENTER_ALWAYS
+        self._border_width = style.LINE_WIDTH
+        self._has_resize_grip = False
+        self._event = Gdk.EventMask.VISIBILITY_NOTIFY_MASK
+        self.set_type_hint(Gdk.WindowTypeHint.DIALOG)
+
+        self._sample_window = Gtk.ScrolledWindow()
+        self._sample_window.set_policy(Gtk.PolicyType.AUTOMATIC,
+                                       Gtk.PolicyType.AUTOMATIC)
+        self._set_decorated(self._decorated)
+        self._set_position(self._position)
+        self._set_border_width(self._border_width)
+        self._set_has_resize_grip(self._has_resize_grip)
+        self._add_events(self._event)
+
+        self.connect('realize', self.__realize_cb)
+        self.connect('key-press-event', self.__key_press_event_cb)
+
+        self._vbox = Gtk.VBox()
+        self.add(self._vbox)
+        self._vbox.show()
+
+        if self._set_title_box:
+            self._title_box = TitleBox(title)
+            self._title_box.close_button.connect(
+                'clicked',
+                self.__close_button_clicked_cb)
+            self._title_box.set_size_request(-1, style.GRID_CELL_SIZE)
+            self._vbox.pack_start(self._title_box, False, True, 0)
+            self._title_box.show()
+
+        width = Gdk.Screen.width() - style.GRID_CELL_SIZE * 5
+        height = Gdk.Screen.height() - style.GRID_CELL_SIZE * 3
+        self.set_size_request(width, height)
+        self.show()
+
+    def get_vbox(self):
+        return self._vbox
+
+    def get_scroll_window(self):
+        return self._sample_window
+
+    def __delete_event_cb(self, chooser, event):
+        self.emit('response', Gtk.ResponseType.DELETE_EVENT)
+
+    def __close_button_clicked_cb(self, button):
+        self.destroy()
+
+    def __key_press_event_cb(self, window, event):
+        keyname = Gdk.keyval_name(event.keyval)
+        if keyname == 'Escape':
+            self.destroy()
+
+    def __realize_cb(self, widget):
+        self.set_type_hint(Gdk.WindowTypeHint.DIALOG)
+        window = self.get_window()
+        window.set_accept_focus(True)
+
+        if self._parent_window_xid is not None:
+            display = Gdk.Display.get_default()
+            parent = GdkX11.X11Window.foreign_new_for_display(
+                display, self._parent_window_xid)
+            window.set_transient_for(parent)
+
+    def _set_decorated(self, decorated):
```

Comment:
Can't the sub classes just use the normal "set_decorated" menthods?  Why do we need to keep a copy of the value as "self._decorated"?  Can't you just set with "set_decorated" and get with "get_decorated"?